src/2024h2/optimize-clippy.md

```diff
@@ -0,0 +1,66 @@
+# Optimizing Clippy & linting
+(a.k.a The Clippy Performance Project)
+
+| Metadata |                                                              |
+| -------- | ------------------------------------------------------------ |
+| Owner(s) | [@blyxyas][blyxyas]                                          |
+| Teams    | [Clippy team][clippy]                                                      |
+| Status   | WIP                                                          |
+
+## Summary
+
+This is the formalization and documentation of the Clippy Performance Project, a project first talked about on [Zulip, July 2023](https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/Clippy's.20performance). As the project consists of several points and is ever-changing, this document also has a dynamic structure and the team can add points. 
+
+In short, this is an effort to optimize Clippy, and Rust's linting infrastructure with a point of view of making Clippy faster (both on CI/CD pipelines, and on devs' machines)
+
+## Motivation
+
+Clippy can take up to 2.5 times the time that a normal `cargo check` takes, and it doesn't need to be! Taking so long is expensive both in development time, and in real money.
+
+On GitHub Actions, this excessive time can equal the cost of running `cargo check` on a Linux x64 32-cores machine, on a Linux x64 2-cores machine. (A 3.3-times increase).
```

Comment:
I find this last sentence a bit confusing. Is this saying that the extra time that `cargo clippy` adds on top of `cargo check` is the time of another `cargo check` execution? Where does the 3.3 come from and how does it compare to the 2.5 mentioned earlier?

Reply:
I knew this would happen, but I still wanted to add those numbers. Their origins are indeed confusing. The time numbers come from Johan Andersson, who in an email conversation talking about the speed problems of Clippy we chatted about the time difference of `cargo clippy` & `cargo check`.

I just made some relations between those numbers and the prices listed [here](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions) (Billing for Gitub Actions).

What the phrase is talking about, is that, because Clippy takes so much, running it in a PR would equal the price of running `cargo check` on a 32-core machine (following Github's billing)